src/plugins/intel_cpu/src/nodes/executors/acl/acl_common_executor.cpp

```diff
@@ -0,0 +1,123 @@
+                                arm_compute::DataType& dataType,
+                                arm_compute::DataLayout& dataLayout) {
+    dataType = precisionToAclDataType(memoryPtr->getPrecision());
+    dataLayout = getAclDataLayoutByMemoryDesc(memoryPtr->getDescPtr());
+    if (dataType != arm_compute::DataType::UNKNOWN) {
+        auto collapsed_dims = collapse_dims_to_max_rank(memoryPtr->getStaticDims(), attrs.maxDimsShape);
+        tensorShape = shapeCast(collapsed_dims);
+        if (attrs.hasLayoutTypeNHWC) {
+            changeLayoutToNH_C({&tensorShape});
+        }
+    }
+}
+
+static ACLInfo initTensorInfo(const arm_compute::TensorShape& tensorShape,
+                              const arm_compute::DataType& dataType,
+                              const arm_compute::DataLayout& dataLayout) {
+    ACLInfo aclMemoryInfo = nullptr;
+    if (dataType != arm_compute::DataType::UNKNOWN) {
+        aclMemoryInfo = std::make_shared<arm_compute::TensorInfo>(
+                tensorShape, 1,
+                dataType,
+                dataLayout);
+    }
+    return aclMemoryInfo;
+}
+
+static ACLMemory initTensor(const ACLInfo& aclMemoryInfo) {
+    ACLMemory aclMemory = nullptr;
+    if (aclMemoryInfo) {
+        aclMemory = std::make_shared<arm_compute::Tensor>();
+        aclMemory->allocator()->init(*aclMemoryInfo);
+    }
+    return aclMemory;
+}
+
+bool ACLCommonExecutor::update(const MemoryArgs &memory) {
+    // Initialize ACL tensors params
+    ACLMemoryShapes  aclMemoryShapes;
+    ACLMemoryTypes   aclDataType{};
+    ACLMemoryLayouts aclDataLayout{};
+    for (auto& cpu_mem_ptr : memory) {
+        const ACLArgs index = argConvert.at(cpu_mem_ptr.first);
+        initACLTensorParams(cpu_mem_ptr.second, aclTensorAttrs,
+                            aclMemoryShapes[index],
+                            aclDataType[index],
+                            aclDataLayout[index]);
+    }
+
+    // Update ACL tensors shapes
+    updateTensorsShapes(aclMemoryShapes);
+
+    // Initialize arm_compute::TensorInfo objects
+    ACLMemoryInfo aclMemoryInfos;
+    for (int i = 0; i < ACLArgs::COUNT_OF_ARGS; i++) {
+        aclMemoryInfos[i] = initTensorInfo(aclMemoryShapes[i], aclDataType[i], aclDataLayout[i]);
+    }
+
+    // Validate arm_compute::TensorInfo objects for specific ACL function
+    auto tensorsInfoValidateStatus = validateTensorsInfo(aclMemoryInfos);
+    if (!tensorsInfoValidateStatus) {
+        DEBUG_LOG("ACL operator validation was failed: ", tensorsInfoValidateStatus.error_description());
+        return false;
+    }
+
+    // Initialize arm_compute::Tensor objects
+    for (int i = 0; i < ACLArgs::COUNT_OF_ARGS; i++) {
+        aclMemoryTensors[i] = initTensor(aclMemoryInfos[i]);
+    }
+
+    // Configure arm_compute::IFunction object
+    configureThreadSafe([&] {
+        iFunction = configureFunction(aclMemoryTensors);
+    });
+    return true;
+}
+
+void ACLCommonExecutor::execute(const MemoryArgs &memory) {
```

Comment:
I propose to leave a todo regarding the fact, that actually it should be enough to import_memory just once in scope of "update()" method, but it is not working for some reason and should be investigated.